Source/init.cpp

```diff
@@ -145,19 +122,55 @@ void init_archives()
 		LogVerbose("MPQ search paths:{}", message);
 	}
 
-#ifndef __ANDROID__
-	// Load devilutionx.mpq first to get the font file for error messages
-	devilutionx_mpq = LoadMPQ(paths, "devilutionx.mpq");
-#endif
-	font_mpq = LoadMPQ(paths, "fonts.mpq"); // Extra fonts
+	return paths;
+}
 
-	if (strcasecmp("en", sgOptions.Language.szCode) != 0 || strlen(sgOptions.Language.szCode) != 2) {
+void init_language_archives(std::vector<std::string> &paths)
+{
+	string_view code = *sgOptions.Language.code;
+	if (code != "en") {
 		char langMpqName[10] = {};
-		CopyUtf8(langMpqName, sgOptions.Language.szCode, sizeof(langMpqName));
+		CopyUtf8(langMpqName, code.data(), sizeof(langMpqName));
 
 		strncat(langMpqName, ".mpq", sizeof(langMpqName) - strlen(langMpqName) - 1);
 		lang_mpq = LoadMPQ(paths, langMpqName);
 	}
+}
+
+} // namespace
+
+void init_cleanup()
+{
+	if (gbIsMultiplayer && gbRunGame) {
+		pfile_write_hero(/*writeGameData=*/false, /*clearTables=*/true);
+	}
+
+	spawn_mpq = std::nullopt;
+	diabdat_mpq = std::nullopt;
+	hellfire_mpq = std::nullopt;
+	hfmonk_mpq = std::nullopt;
+	hfbard_mpq = std::nullopt;
+	hfbarb_mpq = std::nullopt;
+	hfmusic_mpq = std::nullopt;
+	hfvoice_mpq = std::nullopt;
+	lang_mpq = std::nullopt;
+	font_mpq = std::nullopt;
+	devilutionx_mpq = std::nullopt;
+
+	NetClose();
+}
+
+void init_archives()
+{
+	auto paths = GetMPQSearchPaths();
+
+#ifndef __ANDROID__
+	// Load devilutionx.mpq first to get the font file for error messages
+	devilutionx_mpq = LoadMPQ(paths, "devilutionx.mpq");
+#endif
+	font_mpq = LoadMPQ(paths, "fonts.mpq"); // Extra fonts
+
+	init_language_archives();

```

Comment:
We already have `paths` here so let's just:

```suggestion
	init_language_archives(paths);
```

Reply:
That was a careless mistake.  😞
I added the `init_language_archives` overload explicitly for this use case.